feat/deprecate mgno

4rgon4ut opened the pull request:
This PR addresses deprecation of wrapper contract (`GNO --> mGNO`) from deposit and staking process on `Chiado` testnet (and on Gnosis chain mainnet in the future).


Changes:
- feat(deposit): deprecate wrapper contract intermediate call
- chore(constants): remove wrapper address

4rgon4ut commented:
@alebanzas Hello, could you have a look? 
Btw do you have any canonical/specific linting conventions? I just added default prettier config here

filoozom commented:
Hi, this is going to be a bit painful, but could you remove all the linting / formatting in this PR (or open a new one that only does that specifically)?

The idea is that the diff for this feature change should be simple to parse and review, and with all the formatting changes it's quite hard to tell what actually changed in the implementation.

4rgon4ut commented:
> @alebanzas Hello, could you have a look? Btw do you have any canonical/specific linting conventions? I just added default prettier config here

addressed

filoozom commented on src/hooks/use-deposit.js:
```diff
@@ -160,7 +164,7 @@ function useDeposit(wallet, network, tokenInfo) {
           let tx = null
           try {
-            tx = await token.transferAndCall(network.addresses.wrapper, depositAmountBN, data)
+            tx = await token.transferAndCall(network.addresses.deposit, depositAmountBN, data)
```
Same here 🙂

4rgon4ut commented:
> Hi, this is going to be a bit painful, but could you remove all the linting / formatting in this PR (or open a new one that only does that specifically)?
> 
> The idea is that the diff for this feature change should be simple to parse and review, and with all the formatting changes it's quite hard to tell what actually changed in the implementation.

Done

4rgon4ut commented on src/hooks/use-deposit.js:
```diff
@@ -72,7 +72,11 @@ function useDeposit(wallet, network, tokenInfo) {
 
     const wc = newDeposits[0].withdrawal_credentials
-    const isBatch = newDeposits.every(d => d.withdrawal_credentials === wc)
+    let isBatch = false;
+
+    if (newDeposits.length > 1) {
+      isBatch = newDeposits.every(d => d.withdrawal_credentials === wc)
+    }
```
The point is that this line
```js
newDeposits.every(d => d.withdrawal_credentials === wc)
```
always returns `true` even for single deposit in a list. So, this check is necessary to not process single deposits as batches.

filoozom replied:
I feel like there's a bug with the batching thing, because the code implies that it's possible to deposit stake for different withdrawal credentials, but I don't see how this could happen here: https://github.com/gnosischain/deposit-contract/blob/master/contracts/SBCDepositContract.sol#L100-L124

I'll have to investigate this a bit more.

filoozom replied:
Actually @4rgon4ut you should revert this. The naming is a bit confusing, but with one deposit it should go in the batch part. The `isBatch === false` part is when there are deposits with different withdrawal credentials, as these actually require multiple transactions.

So please revert this change 🙂

4rgon4ut replied:
@filoozom done

4rgon4ut commented:
> Have you tested this on Chiado? I see 3 new deposits, were those yours?

Yes, I tested this on `Chiado` but I have done only 2 of them
